hledger-stockquotes-0.1.2.0 fails to compile with new hledger-lib release

```
/var/stackage/work/unpack-dir/unpacked/hledger-stockquotes-0.1.2.0-0f43170c0890b851da75c4a94caee40cd638b522a0c41336e7e81d736db13438/src/Hledger/StockQuotes.hs:66:40: error:
    • Couldn't match type ‘transformers-0.5.6.2:Control.Monad.Trans.Except.ExceptT
                             String IO’
                     with ‘IO’
      Expected: IO (Either [Char] Journal)
        Actual: transformers-0.5.6.2:Control.Monad.Trans.Except.ExceptT
                  String IO Journal
    • In the second argument of ‘(<$>)’, namely
        ‘readJournalFile definputopts journalPath’
      In a stmt of a 'do' block:
        journal <- either error id
                     <$> readJournalFile definputopts journalPath
      In the expression:
        do journal <- either error id
                        <$> readJournalFile definputopts journalPath
           currentTime <- getCurrentTime
           let commodities
                 = filter (...)
                     $ M.keys (jcommodities journal)
                         <> M.keys (jinferredcommodities journal)
               dates = map tdate $ jtxns journal
               ....
           return (L.sort $ L.nub commodities, minDate, maxDate)
   |
66 |     journal     <- either error id <$> readJournalFile definputopts journalPath
   |                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Just like #6605, will drop `hledger-stockquotes` from the snapshot to minimize blast radius.

CC @prikhi

--- a/build-constraints.yaml
+++ b/build-constraints.yaml
@@ -4418,7 +4418,7 @@ packages:
         - stack-templatizer
         - immortal-queue
         - wai-middleware-clacks
-        - hledger-stockquotes < 0 # 0.1.2.0 compile fail https://github.com/commercialhaskell/stackage/issues/6607
+        - hledger-stockquotes
         - bnb-staking-csvs
         - solana-staking-csvs
         - cointracking-imports

